### DISCUSSION TITLE
AddFluentMigratorCore requires FluentMigrator.Runner

### DISCUSSION THREAD

**Discussion by fairking** [fluentmigrator/fluentmigrator#1584] [Ideas]
I am not sure if it's a big deal. I am using Sqlite in my project. Also I use dependency injection (as I found it's kind of mandatory to execute migration scripts).
So I added `FluentMigrator` and `FluentMigrator.Runner.Sqlite` nugets. Then I have found that there is no way to use it without actually adding `FluentMigrator.Runner` which contains all database implementations I don't need.
From the other side why do we need `FluentMigrator.Runner.Sqlite` then if you have to add `FluentMigrator.Runner` anyway?

The question is: should we move `AddFluentMigratorCore()` and `IMigrationRunner` to `FluenMigrator.Runner.Core` project? So I can simply use `FluentMigrator.Runner.Sqlite`?

**Comment by jzabroski** [comment]
I would be open to that change. I am hoping to ship a light 4.0 release in the next week based on @mattbrailsford changes that focus around Sqlite.  I'm trying to re-think how I manage this project going forward, and I think I have been to hesitant to make breaking changes and it slows progress.

**Reply by fairking** [reply]
@jzabroski If I had some [free time](https://gitlab.com/fairking/napos) I would help you.

**Comment by fairking** [comment]
@jzabroski I would suggest to think about all experience, create a new project and move gradually stuff from the old one starting with the most used database. You don't have to write a new code, just to place puzzles (classes) you already have. IMHO.

**Comment by Iceeman76** [comment]
Any news on this? I have the same problem.. want to use FluentMigrator in a Blazor WASM project but it is not possible right now because of all the database implementations that ships with it.
